src/csharp/Grpc.Tools/build/_protobuf/Google.Protobuf.Tools.targets

```diff
@@ -127,15 +128,23 @@
     </FindUnderPath>
     <ItemGroup>
       <!-- Files with explicit metadata. -->
-      <Protobuf_Compile Include="@(Protobuf->HasMetadata('ProtoRoot'))" />
+      <Protobuf_Rooted Include="@(Protobuf->HasMetadata('ProtoRoot'))" />
       <!-- In-project files will have ProtoRoot='.'. -->
-      <Protobuf_Compile Include="@(_Protobuf_NoRootInProject)">
+      <Protobuf_Rooted Include="@(_Protobuf_NoRootInProject)">
         <ProtoRoot>.</ProtoRoot>
-      </Protobuf_Compile>
+      </Protobuf_Rooted>
       <!-- Out-of-project files will have respective ProtoRoot='%(RelativeDir)'. -->
-      <Protobuf_Compile Include="@(_Protobuf_NoRootElsewhere)">
+      <Protobuf_Rooted Include="@(_Protobuf_NoRootElsewhere)">
         <ProtoRoot>%(RelativeDir)</ProtoRoot>
-      </Protobuf_Compile>
+      </Protobuf_Rooted>
+    </ItemGroup>
+  </Target>
+
+  <!-- Select files that should be compiled. -->
+  <Target Name="_Protobuf_SelectFiles">
```

Comment:
Minor nit and I think I'm splitting hairs here, but I think it makes more sense to have _Protobuf_SelectFiles depend on _Protobuf_SetProtoRoot instead of Protobuf_Compile logically speaking.

Reply:
Only thought about the main target but you are right that it relies on the root being populated. So I should definitely adjust it!

Reply:
Fixed in latest commit.